### ISSUE TITLE
The [Bug Report] Upload component before-upload returns false, and all uploads are cancelled

### ISSUE THREAD

**Issue by Darkbish** [ElemeFE/element#7076]
### Element UI version
1.4.4

### OS/Browsers version
win 10 1703 / chrome 61.0.3163.91

### Vue version
2.4.4

### Reproduction Link
[https://jsfiddle.net/4heo5to9/3/](https://jsfiddle.net/4heo5to9/3/)

### Steps to reproduce
最多只能上传3张图片，index >= 3时，beforeUpload 返回 false，然后所有的上传就都cancelled

### What is Expected?
index >= 3时，该index所指向的文件取消上传，但是不影响 index < 3的文件

### What is actually happening?
index >= 3的文件没有上传，但是 index < 3 的文件也取消了上传

**Comment by Leopoldthecoder** [comment]
Fixed in #7077

**Comment by Darkbish** [comment]
An advice to Upload compenent:
> 
```
watch: {
    fileList: {
      immediate: true,
      handler(fileList) {
        this.uploadFiles = fileList.map(item => {
          item.uid = item.uid || (Date.now() + this.tempIndex++);
          item.status = 'success';
          return item;
        });
      }
    }
  },
```
when `fileList` has changed, all `uid` always re-generate. I think the `uid` should not re-generate and keep it unique.
